core/src/main/java/org/infinispan/reactive/publisher/impl/LocalClusterPublisherManagerImpl.java

```diff
@@ -169,29 +168,36 @@ IntSet handleNullSegments(IntSet segments) {
       if (transformer instanceof InjectableComponent) {
          ((InjectableComponent) transformer).inject(componentRegistry);
       }
+
       if (invocationContext == null || invocationContext.lookedUpEntriesCount() == 0) {
-         return localPublisherManager.entryPublisher(handleNullSegments(segments), keysToInclude, null, includeLoader,
-               DeliveryGuarantee.AT_MOST_ONCE, transformer);
+         return localPublisherManager.entryPublisher(handleNullSegments(segments), keysToInclude, null,
+               includeLoader, DeliveryGuarantee.AT_MOST_ONCE, transformer);
       }
-      SegmentPublisherSupplier<R> cachePublisher = localPublisherManager.entryPublisher(handleNullSegments(segments), keysToInclude, null,
-            includeLoader, DeliveryGuarantee.AT_MOST_ONCE, transformer);
 
-      Flowable<CacheEntry<K, V>> entryFlowable = entryPublisherFromContext(invocationContext, segments, keyPartitioner,
-            keysToInclude);
+      SegmentPublisherSupplier<R> cachePublisher = localPublisherManager.entryPublisher(handleNullSegments(segments),
+            keysToInclude, k -> invocationContext.lookupEntry(k) != null, includeLoader, DeliveryGuarantee.AT_MOST_ONCE, transformer);
```

Comment:
IMO it's not worth instantiating a lambda, let's just un-deprecate `InvocationContext.getLookedUpEntries()` :)

Reply:
Sure, I can do that.

Reply:
I left the publisher methods as I was liking those a bit :)